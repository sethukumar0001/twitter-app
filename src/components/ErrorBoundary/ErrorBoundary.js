import React from 'react';
import { Typography, Grid } from '@material-ui/core';

class ErrorBoundary extends React.Component {
  state = {
    error: null,
    info: null,
  };

  componentDidCatch(error, info) {
    this.setState({ error, info });
  }

  render() {
    const { error, info } = this.state;

    if (!error) {
      return this.props.children;
    }

    return (
      <Grid container justify="center">
        <Grid item xs={11} sm={6} md={4}>
          <Typography variant="display3">Woops :(</Typography>
          <Typography paragraph variant="subheading">
            Looks like we have run into troubles
          </Typography>
          <pre
            style={{
              whiteSpace: 'pre-wrap',
              overflow: 'scroll',
              height: 300,
            }}
          >
            {info.componentStack}
          </pre>
        </Grid>
      </Grid>
    );
  }
}

export default ErrorBoundary;
