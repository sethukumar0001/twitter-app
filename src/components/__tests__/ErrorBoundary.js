import React from 'react';
import renderer from 'react-test-renderer';
import ErrorBoundary from '../ErrorBoundary';

describe('ErrorBoundary', () => {
  it('should render correctly', () => {
    const component = renderer.create(
      <ErrorBoundary>
        <h1>Heading</h1>
      </ErrorBoundary>,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render with error', () => {
    const Broken = () => {
      throw new Error('boom!');
    };
    console.error = jest.fn();

    const component = renderer.create(
      <ErrorBoundary>
        <Broken />
      </ErrorBoundary>,
    );

    expect(console.error).toBeCalled();
    expect(component.toJSON()).toMatchSnapshot();
  });
});
