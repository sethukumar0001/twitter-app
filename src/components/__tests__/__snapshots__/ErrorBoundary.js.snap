// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary should render correctly 1`] = `
<h1>
  Heading
</h1>
`;

exports[`ErrorBoundary should render with error 1`] = `
<div
  className="MuiGrid-container-1 MuiGrid-justify-xs-center-18"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-11-40 MuiGrid-grid-sm-6-49 MuiGrid-grid-md-4-61"
  >
    <h1
      className="MuiTypography-root-98 MuiTypography-display3-100"
    >
      Woops :(
    </h1>
    <p
      className="MuiTypography-root-98 MuiTypography-subheading-105 MuiTypography-paragraph-116"
    >
      Looks like we have run into troubles
    </p>
    <pre
      style={
        Object {
          "height": 300,
          "overflow": "scroll",
          "whiteSpace": "pre-wrap",
        }
      }
    >
      
          in Broken (at ErrorBoundary.js:22)
          in ErrorBoundary (at ErrorBoundary.js:21)
    </pre>
  </div>
</div>
`;
